classes/core/PKPPageRouter.inc.php

```diff
@@ -396,10 +396,20 @@ function url($request, $newContext = null, $page = null, $op = null, $path = nul
 	/**
 	 * @copydoc PKPRouter::handleAuthorizationFailure()
 	 */
-	function handleAuthorizationFailure($request, $authorizationMessage) {
+	function handleAuthorizationFailure(
+		$request,
+		$authorizationMessage,
+		array $messageParams = []
+	) {
 		// Redirect to the authorization denied page.
 		if (!$request->getUser()) Validation::redirectLogin();
-		$request->redirect(null, 'user', 'authorizationDenied', null, array('message' => $authorizationMessage));
+		$request->redirect(
+			null,
+			'user',
+			'authorizationDenied',
+			null,
+			['message' => __($authorizationMessage, $messageParams)]
```

Comment:
@NateWr Using `__` here makes sense? I'm in doubt about this.

Reply:
I think your doubt is correct. In the `PKPPageRouter`, the `$authorizationMessage` is passed as a locale key to the query string. So what happens is that this method redirects the user to a URL like this:

```
https://site.com/journal1/user/authorizationDenied?message=...
```

That then gets routed to the UserHandler, which picks up the locale key [here](https://github.com/pkp/pkp-lib/blob/master/pages/user/PKPUserHandler.inc.php#L83-L91). Then a plain message template is loaded that [translates the locale key](https://github.com/pkp/pkp-lib/blob/master/templates/frontend/pages/message.tpl#L21-L23).

So I think what we need to do is pass the message params as a param to the redireect like this:

```php
$request->redirect(
	null,
	'user',
	'authorizationDenied',
	null,
	[
		'message' => $authorizationMessage,
		'params' => $messageParams,
	]
);
```

Then in `PKPUserHandler`, get the params, assign them to the template, and use them in the `{translate ...}` function in the message template.